lib/manager/poetry/extract.ts

```diff
@@ -84,3 +85,20 @@ function extractFromSection(
   });
   return deps;
 }
+
+function extractRegistries(pyprojectfile: PoetryFile): string[] {
+  const sources = pyprojectfile.tool.poetry.source;
```

Comment:
Need to wrap this in a try/catch or check existence of pyprojectfile.tool.poetry. Otherwise if someone deletes this section from a file it will cause an exception

Reply:
```suggestion
  const sources = pyprojectfile.tool && pyprojectfile.tool.poetry && pyprojectfile.tool.poetry.source;
```

Note: needs prettierfying